### PR TITLE
podio_class_generator: only write files if their content changed

### PR DESCRIPTION
**Pull request by andresailer**
Avoids spurious re-compilation
BEGINRELEASENOTES
- podio_class_generator:  only write files if their content changed

ENDRELEASENOTES

### CODE REVIEW

File: python/podio_class_generator.py [hunk 1]
```diff
@@ -1034,17 +1033,22 @@ def prepare_vectorized_access(self, classname,members ):
...
-        open(fullname, "w").write(content)
         if self.clang_format:
-          subprocess.call(self.clang_format + [fullname])
+          cfproc = subprocess.Popen(self.clang_format, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+          content = cfproc.communicate(input=content)[0]
+        existing_content = ''
+        if os.path.exists(fullname):
+          existing_content = open(fullname, 'r').read()
+        if existing_content != content:
+          open(fullname, 'w').write(content)
```
Comment (vvolkl) [inline python/podio_class_generator.py]: These lines could be slightly more "pythonic" with a `with` statement: 
```python
with open(fullname, "r+") as _f:
  existing_content = _f.read()
  if existing_content != content:
    _f.write(content)
```

Comment (andresailer) [reply python/podio_class_generator.py]:
```python
#!/usr/bin/env python
content ='foo bar'
fullname = 'temp.txt'
with open(fullname, "r+") as _f:
  existing_content = _f.read()
  if existing_content != content:
    _f.write(content)
```
Errors if temp.txt does not exist
```
$ python testRead.py
Traceback (most recent call last):
  File "testRead.py", line 6, in <module>
    with open(fullname, "r+") as _f:
IOError: [Errno 2] No such file or directory: 'temp.txt'
```
So we would still need one path for exists, one without.

Comment (vvolkl) [reply python/podio_class_generator.py]: That's true, I overlooked that in one case the file won't exist yet. Taking that in consideration I think the code is best as it is!

Comment (andresailer) [reply python/podio_class_generator.py]: There are many things to be cleaned up, especially in view of python3 compatibility (#66)

**Pull request merged**